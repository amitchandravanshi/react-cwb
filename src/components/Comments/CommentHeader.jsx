import React from "react";
import { withStyles } from "@material-ui/core/styles";
import CardHeader from "@material-ui/core/CardHeader";

const styles = theme => {
  return {
    root: {
      padding: ".8rem",
      width: "80%"
    },
    header: {
      display: "flex",
      justifyContent: "space-around",
      alignItems: "baseline"
    },
    contentWrapper: {
      width: "100%",
      display: "flex",
      justifyContent: "flex-start",
      flexWrap: "wrap"
    },
    contentHeader: {
      fontSize: "14px",
      fontWeight: "bold",
      width: "100%"
    },
    contentFooter: {
      fontSize: "12px",
      color: "#ccc"
    }
  };
};

class CommentHeader extends React.Component {
  get titleHeader() {
    const { classes, title, subheader } = this.props;
    return (
      <div className={classes.header}>
        <div className={classes.contentWrapper}>
          <span className={classes.contentHeader}>{title}</span>
          <span className={classes.contentFooter}>{subheader}</span>
        </div>
      </div>
    );
  }
  render() {
    const { classes } = this.props;
    return <div className={classes.root}>{this.titleHeader}</div>;
  }
}

export default withStyles(styles)(CommentHeader);
