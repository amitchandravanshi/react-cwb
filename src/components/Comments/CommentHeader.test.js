import React from "react";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import CommentHeader from "./CommentHeader";

const props = {
  classes: {}
};
describe("<CommentHeader> ", () => {
  it("<CommentHeader> snapshot ", () => {
    const wrapper = mount(<CommentHeader {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
