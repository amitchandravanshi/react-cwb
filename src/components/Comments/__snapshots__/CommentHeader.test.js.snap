// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommentHeader>  <CommentHeader> snapshot  1`] = `
<CommentHeader
  classes={
    Object {
      "contentFooter": "CommentHeader-contentFooter-5",
      "contentHeader": "CommentHeader-contentHeader-4",
      "contentWrapper": "CommentHeader-contentWrapper-3",
      "header": "CommentHeader-header-2",
      "root": "CommentHeader-root-1",
    }
  }
/>
`;
